memos.rb

```diff
@@ -2,23 +2,41 @@
 
 require 'sinatra'
 require 'sinatra/reloader'
-require 'json'
+require 'pg'
 require 'cgi'
 
-def save_file
-  File.open('public/json/data.json', 'w') { |file| JSON.dump(@memos, file) }
+def generate_memos(id)
+  @memos = { 'memo_id': id, 'title': params['title'], 'content': params['content'] }
 end
 
-def load_file
-  File.open('public/json/data.json') { |file| @memos = JSON.parse(file.read) }
+def load_all_memos
+  connect = PG::Connection.new(host: 'localhost', user: 'postgres', dbname: 'memo')
+  @memos = connect.exec('select * from memos order by memo_id asc')
```

Comment:
これだと `@memos` はトップレベルのコンテキストに定義されるので、グローバル変数を定義するのと変わらないことになってしまいます 🙅‍♂️ たとえば `get '/memos'` のところで以下のようにメソッドを呼び出すと、これに起因して意図しない挙動になってしまします。メソッド側では戻り値を返すようにして、呼び出し側でそれを使うようにしましょう。他のメソッドについても同様の変更をおねがいします。

```ruby
get '/memos 'do
  load_all_memos
  load_detail_memo # ここで @memos が上書きされて :index が正しく表示されない
  @title = 'index'
  erb :index
end
```

Reply:
`README.md` のほうにコメントしたスタイルで SQL を書いてみてください 🙏 
```suggestion
  @memos = connect.exec('SELECT * FROM memos ORDER BY memo_id ASC')
```

Reply:
`@memos`に戻り値を代入するように変更しました。[e4691af](https://github.com/tomonariha/memos/pull/2/commits/e4691afa78da244e670f9ab65d3b335752138116)
また、SQL文の書き方を変更しました。[4ed84a1](https://github.com/tomonariha/memos/pull/2/commits/4ed84a15135c89eedc48285ff49a50db0c392d8f)